public/js/components/ObjectInspector.js

```diff
@@ -0,0 +1,169 @@
+"use strict";
+
+const React = require("react");
+const classnames = require("classnames");
+const ManagedTree = React.createFactory(require("./util/ManagedTree"));
+const Arrow = React.createFactory(require("./util/Arrow"));
+const Rep = require("./Rep");
+const { DOM: dom, PropTypes } = React;
+
+// This implements a component that renders an interactive inspector
```

Comment:
Thanks for the high-level description, I find these really helpful at the top of files.